### PR TITLE
feat(TDI-32103):add support hardDelete on salesforce

### PR DESCRIPTION
**Pull request by pyzhou-talend**
**What is the current behavior?** (You can also link to an open issue here)

https://jira.talendforge.org/browse/TDI-32103

**What is the new behavior?**

**Please check if the PR fulfills these requirements**

- [ ] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/9/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/9/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Apyzhou%2FTDI-32103-add_support_for_salesforce_hardDelete)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/10/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/10/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Apyzhou%2FTDI-32103-add_support_for_salesforce_hardDelete)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/14/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/14/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/17/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/17/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Apyzhou%2FTDI-32103-add_support_for_salesforce_hardDelete)

File: components/components-salesforce/components-salesforce-definition/src/main/java/org/talend/components/salesforce/SalesforceOutputProperties.java [hunk 1]
```diff
@@ -41,6 +42,8 @@
 
     public Property<OutputAction> outputAction = newEnum("outputAction", OutputAction.class); // $NON-NLS-1$
 
+    public Property<Boolean> hardDelete = newBoolean("hardDelete");
```
[the remark this answers is not in the document]
Comment (pyzhou-talend) [reply components/components-salesforce/components-salesforce-definition/src/main/java/org/talend/components/salesforce/SalesforceOutputProperties.java]: No. it only shows when bulk load v1 API is using and output operation is Delete.

**Pull request merged**